### ISSUE TITLE
Individual Control Style Pages with Resources and Visual States for UWP/WinUI

### ISSUE THREAD

**Issue by michael-hawker** [MicrosoftDocs/windows-uwp#1870]
There used to be equivalent style pages for each control like [this one for WPF](https://docs.microsoft.com/en-us/dotnet/framework/wpf/controls/progressbar-styles-and-templates) listing out the style info from generic.xaml for the control.  It makes it super easy to understand a control's style and construction as a reference when writing subcontrols using that template.  Searching generic.xaml is a pain (if you remember the path where it's located), and assuming you know it's a thing (as it's not called out in this article either).

Now it seems like all those bing results just point to this general style information guide instead.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 71085aa7-f688-0f7f-af84-431742ca7a2a
* Version Independent ID: 181f13bd-fb4e-3e43-398d-8a2f6e981035
* Content: [XAML styles - Windows UWP applications](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/xaml-styles#feedback)
* Content Source: [windows-apps-src/design/controls-and-patterns/xaml-styles.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/design/controls-and-patterns/xaml-styles.md)
* Product: **uwp**
* Technology: **design**
* GitHub Login: @jwmsft
* Microsoft Alias: **jimwalk**

**Comment by jevansaks** [comment]
We can link out to the github pages where these styles are defined if that would help, or maybe even cross-include the relevant portions into the doc pages. @chigy something to think about for how we could automate this process?

**Comment by chigy** [comment]
@michael-hawker , thank you for your request. We are aware that this is something we currently don't have any sustainable solution for.

That said, it would be helpful if you could help me understand what exactly you are looking for. Are you simply looking for us to extract a copy of style in XAML markup like you linked for WPF or do you want to see the visual and notation on the UI? 

I am asking not because we can do this, but because I wonder how valuable the copy of XAML style would be for those who are not too familiar with XAML and those who can use that info, while it is not ideal, can still go to generic.xaml?

**Comment by michael-hawker** [comment]
Hi @chigy, the main thing is that searching through the massive generic.xaml is a pain, as there's generally the control name used as a lot of prefixes for other resources in that file as well, so outside of remembering where the file is to dig up, it takes a bit to search through.  It was much faster to just bing `uwp listview style` and pull up the result as a reference online.

The main thing I used the docs for in the past were the following:
  
  - Copy the style for to modify in my own application (as the VS/Blend feature that does this isn't reliable)
  - Reference the original in my modified style/control to debug
  - Investigate the composition of the control to understand how to modify it's behavior or access sub-components when implementing a control inheriting from a base control
  - Look at the theme/style constants broken out in the doc upper section to understand which resources I have easy access to to override/modify
  - Look at the list of Visual States to understand/modify control behavior

Hope that helps!

**Comment by chigy** [comment]
@mijacobs , is this something we could potentially use tagging of the Controls Gallery file?

**Comment by mijacobs** [comment]
Are you talking about using the snippet system to tag areas of generic.xaml and then having pages for them in the docs?

**Comment by chigy** [comment]
@mijacobs , yes that's what I meant.

**Comment by mijacobs** [comment]
Yes, that would work. It would be a bit of work to create them initially, but then they'd update automatically whenever they style sheet was updated. It might complicate updating generic.xaml, though, because you'd need to preserve the snippet tags so you couldn't just overwrite the file.

**Comment by chigy** [comment]
OK, let me figure out the priority of doing this.

**Comment by Depechie** [comment]
hey @chigy any news on this?

**Comment by chigy** [comment]
@Depechie , currently unfortunately, there is not a priority on this item. That said, I am curious if we were to provide info like this, I'd be curious to find out how you would use this.

**Comment by Depechie** [comment]
Not sure where that questions comes from, but the answer is already in the list : https://github.com/MicrosoftDocs/windows-uwp/issues/1870#issuecomment-523609902

When we create apps and see that a control behaves in a certain way, often we want to retemplate the style to edit the VisualStates. Without an easy reference to look and to copy this is hard work.

**Comment by chigy** [comment]
> Not sure where that questions comes from, but the answer is already in the list : [#1870 (comment)](https://github.com/MicrosoftDocs/windows-uwp/issues/1870#issuecomment-523609902)
> 
> When we create apps and see that a control behaves in a certain way, often we want to retemplate the style to edit the VisualStates. Without an easy reference to look and to copy this is hard work.

@Depechie , thanks. I wasn't sure if you had additional cases you were after.

**Comment by robloo** [comment]
Adding docs for the default control templates/styles is really important in my opinion. Copying some comments from #2772:

 * Concerning the discussion about re-templating in general: I notice a while ago ALL the documentation seemed to disappear for re-templating specific controls. You used to be able to go to Microsoft docs online and see the default template as well as which parts were required. This documentation for each control is really critical for developers to understand specific quirks and concerns when re-templating is required for any given control.
 * Aside from general quirks/instructions. The most useful documentation is also 'why' template parts were added. It should be explained someplace what is going on internally that require these. In some cases, the part's backgrounds/fill are automatically calculated and set. That should all be explained. It should also be explained which template parts are designed to be safely excluded.
 * All controls, should declare their parts in code. Then the XAML could automatically be added to docs and the parts automatically detected. This ensures everything stays in sync.
 * The areas of the documentation I'm more concerned are notes from the original control authors. Those authors should provide quirks/instructions for downstream developers that may have to re-template. These notes shouldn't really change and don't need to be kept up-to date with automated tooling.

**Comment by michael-hawker** [comment]
Also see the Style Migration request here: https://github.com/microsoft/microsoft-ui-xaml/discussions/7497 (separate but also important that could be worked into these types of guides).

This was also called out again recently on Twitter here by @nickrandolph: https://twitter.com/thenickrandolph/status/1573122578151247873?s=20&t=eiGoprJGS6R1rajrA4ulpA

![image](https://user-images.githubusercontent.com/24302614/192057515-ac5afe9b-0342-4ec7-b788-2dbf55a3a234.png)

Updating the title of this ask to be broader as well.

**Comment by GrantMeStrength** [comment]
Closing — this issue references a UWP page (`windows-apps-src/`) that is now maintained in the [windows-uwp](https://github.com/MicrosoftDocs/windows-uwp) repository, not this one. If the issue is still relevant, please re-file it there. Thank you!
